dot/types/header.go

```diff
@@ -75,9 +87,14 @@ func (bh *Header) DeepCopy() *Header {
 		cp.Number = new(big.Int).Set(bh.Number)
 	}
 
-	if len(bh.Digest) > 0 {
-		cp.Digest = make([]DigestItem, len(bh.Digest))
-		copy(cp.Digest[:], bh.Digest[:])
+	if len(bh.Digest.Types) > 0 {
+		cp.Digest = NewDigest()
+		for _, d := range bh.Digest.Types {
+			err := cp.Digest.Add(d.Value())
+			if err != nil {
+				panic(err)
```

Comment:
we shouldn't have panics in the code.  Change the function sig of `DeepCopy` to return an error and handle it on the calling side.